lib/fluent/plugin/out_splunk_hec.rb

```diff
@@ -103,7 +108,11 @@ def format_event(time, record)
       msg['time'] = time if @use_fluentd_time
 
       # metadata
-      msg['sourcetype'] = @sourcetype if @sourcetype
+      if record[@sourcetype_key]
+        msg['sourcetype'] = @remove_sourcetype_key ? record.delete(@sourcetype_key) : record[@sourcetype_key]
+      elsif @default_sourcetype
+        msg['sourcetype'] = @default_sourcetype
```

Comment:
Appending `if @default_sourcetype` to this line seems to be needed since `@default_sourcetype` can be `nil`.

Reply:
No it can't since it is checked on row 113? Or am I missing something else...

Reply:
ah, my bad. I missed L113.